### DISCUSSION TITLE
Quarkus-Rest or Resteasy-Classic

### DISCUSSION THREAD

**Discussion by hamburml** [quarkusio/quarkus#43500] [Community]
I know quarkus-rest (or resteasy-reactive) is matured overtime. Do you have any numbers how often quarkus-rest is used in comparison to resteasy-classic?

**Comment by cescoffier** [comment]
@maxandersen and @brunobat should have the number. 

But, anyway, we **strongly** recommend Quarkus REST.

**Reply by brunobat** [reply]
Yes, in the last 30 days, `quarkus-rest` was used 45% more than `quarkus-resteasy`.
Also, `quarkus-rest-jackson` was used 56% more than `quarkus-resteasy-jackson`.
`quarkus-rest-jackson` was the 3rd most used extension, only behind `quarkus-arc` and `quarkus-smallrye-openapi`.

**Reply by hamburml** [reply]
Thanks for the numbers!

**Reply by apollo7321** [reply]
Out of curiosity, since https://quarkus.io/guides/rest mentions at the top:

> This is the reference guide for Quarkus REST. For a more lightweight introduction, please refer to the [Writing JSON REST services guides](https://quarkus.io/guides/rest-json).

quarkus-rest or quarkus-jackson/quarkus-jsonb?
As far as I can tell quarkus-rest is heavier but brings things like reactive libs and virtual-threads with it.

But so far I am unsure what the difference is.

**Reply by cescoffier** [reply]
It's not heavier, it's actually lighter. 

In your comment, you mentioned JSON-B. We recommend quarkus-rest-jackson.

**Reply by apollo7321** [reply]
@cescoffier thank you for your answer.

`quarkus-rest` is lighter than `quarkus-rest-jackson`?

That is interesting, I thought it bundles less than `quarkus-rest` will.

I have no preference for jackson nor jsonb. In my project I just use `quarkus-rest`, but I wanted to find out what the difference are and came across this discussion.

**Reply by gsmet** [reply]
> quarkus-rest is lighter than quarkus-rest-jackson?

`quarkus-rest` is the base library for anything REST. If you want JSON support with Jackson then you add `quarkus-rest-jackson` (which also will bring `quarkus-rest` anyway).

If you don't need to serialize/deserialize JSON, then just use `quarkus-rest`.

If you serialize JSON, you have the choice between:
- Jackson with `quarkus-rest-jackson`
- JSON-B with `quarkus-rest-jsonb`

We recommend to use the former if you don't stricly want the latter.

**Reply by hamburml** [reply]
> quarkus-rest is the base library for anything REST

Only when not using [resteasy](https://resteasy.dev/), right? I am still hesitant to switch to it because lots of colleagues had issues with (for example) general exception mappers which were located in microprofile libs - they worked fine with resteasy but didnt worked anymore with quarkus-rest. They needed to switch to the `@ServerExceptionMapper` annotation but now the lib is not compatible to resteasy anymore. To be fair this was a year ago or so, so these issues may be resolved.

I asked them to make a simple reproducer and provide them here - not sure if this was done. I do not think so tbh.
